dts/bindings/input/zephyr,lvgl-button-input.yaml

```diff
@@ -0,0 +1,47 @@
+# Copyright 2023 Fabian Blatz <fabianblatz@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  LVGL button indev pseudo-device
+
+  Listens for button input events and routes the
+  lv_indev_data_t to the underlying button lv_indev_t managed by LVGL.
+
+  Needs to be associated to a specific device to listen for events
+  from that device. Example configuration:
+
+  pointer {
+          compatible = "zephyr,lvgl-button-input";
+          input = <&buttons>;
+          input-codes = <INPUT_KEY_0>;
+          lvgl-key = <10>;
+          x-coordinate = <120>;
+          y-coordinate = <120>;
```

Comment:
hey, sorry caught this one late, trying to understand the LVGL button device a bit better. Would it make sense to have these three as arrays so that `input-codes` map to pairs of `x-coordinates` `y-coordinates`? Then about `lvgl-key`, does that really have to be an `LV_KEY_*`? From what I understand that should instead be the index of the x,y pair passed `points_array` on device registration.

So I'm thinking this should be specified as, for example:
```
          input-codes = <INPUT_KEY_0, INPUT_KEY_0>;
          x-coordinate = <120, 110>;
          y-coordinate = <120, 110>;
```

Where the pairs `120, 120` and `110, 110` would be registered and `lvgl_button_process_event` would just send, say key offset 0, 1 etc...

Reply:
Ahhhhh, right. I thoroughly misunderstood that one. I thought that the logic of the lvgl indev was that a key down event with the `LV_KEY_*` at all of the locations was raised. That was why I intially had only support for one point added because I did not see why a `multiclick` would make sense. Rather the `btn_id` is basically indexing into the the coordinates array. This also has the benefit that you really only need one pseudo device for X physical buttons. Neat.

Thanks a lot for raising this  :heart: . Maybe I'll open a PR upstream and a short paragraph explaining this :^)

Reply:
Yeah I mixed them up as well, I find the doc unclear, ended up checking the code directly: https://github.com/zephyrproject-rtos/lvgl/blob/8a6a2d1d29d17d1e4bdc94c243c146a39d635fdd/src/core/lv_indev.c#L768

Reply:
Added the implementation according to your suggestion, works like a charm :^)

Reply:
Raised a smol PR upstream to update docs :^):
[lvgl/pull/4478](https://github.com/lvgl/lvgl/pull/4478)